standards/travis-ci.md

```diff
@@ -102,6 +102,44 @@ before_script:
 
 The installation of Chrome would still be required even if chrome-headless is used by `karma` or other test suites.
 
+### Failure to Decrypt Environmental Variables
+
+If it seems your encrypted environmental variables aren't being set in builds and/or you see something like the following in your build log, this section is for you:
+```
+Setting environment variables from .travis.yml
+$ export CpxjaQ=[secure]
+...
+```
+
+One reason Travis will fail to properly decrypt environmental variables is if there's a mismatch between the Travis endpoint used to encrypt the values and the Travis endpoint used to run builds. Travis is in the process of moving from travis-ci.org to travis-ci.com. All builds will eventually migrate to the latter. Our repos are currently split between them, with most of the newer Travis integrations existing on `.com`.
+
+**To determine whether `.org` or `.com` is correct for a given repo**, simply check the Travis URL of any attempted build.
+
+**To determine which endpoint is actually used to encrypt on your local machine**, check `~/.travis/config.yml`. For example:
+
+```
+endpoints:
+  https://api.travis-ci.org/:
+    access_token: [snip]
+  https://api.travis-ci.com/:
+    access_token: [snip]
+repos:
+  NYPL-discovery/discovery-front-end:
+    endpoint: https://api.travis-ci.org/
+  NYPL-discovery/schemaservice:
+    endpoint: https://api.travis-ci.com/
+```
+
+The `endpoints` section indicates I've authenticated against both `.org` and `.com` at some point. Under `repos`, we see that builds for discovery-front-end are expected to be run on `.org` whereas builds for schemaservice are expected to be run on `.com`. Consequently any time I run `travis encrypt ...` locally for either of those repos, the encryption will be tied to those endpoints.
+
+To correct an endpoint mismatch locally, re-run your `travis encrypt ...` command with ` --com` or ` --org` at the end. (This will add the missing entry to your `~/.travis/config.yml` for subsequent calls.)
```

Comment:
"endpoint mismatch" — might be useful to explain how that actually comes about. I.e., the `travis encrypt` command was run for the wrong domain (either explicitly, because of the default setting, or because of logging in to the wrong domain — as far as I understand it). I'm not confident that the words I just wrote are correct, _but_ that illustrates why I'd appreciate a little more clarification :-)

Reply:
Something like this?

> When you run `travis encrypt ...`, the command makes a guess about whether the integration will be run on `.org` or `.com`. It makes that guess based on the contents of the `repos` section described above and, failing a match there, falls back on your default endpoint (see notes on `travis endpoint ...` below). When `travis encrypt` defaults to the wrong endpoint, correct that by re-running your `travis encrypt ...` command with ` --com` or ` --org` at the end. (This will add the missing entry to your `~/.travis/config.yml` for subsequent calls.)

If my understanding is right, I wonder why `travis encrypt` is unable to determine *which* endpoint to use via its own api (i.e. ping the travis-ci.org repo endpoint to see if the repo has an integration there. Failing that, ping the `.com`...) This is all oddly manual

Reply:
Yeah, that looks great @nonword — thank you!